syft/frameworks/torch/nn/rnn.py

```diff
@@ -185,106 +185,77 @@ def __init__(
         # Link to issue: https://github.com/OpenMined/PySyft/issues/2500
 
         # Build RNN layers
-        self.rnn_forward = nn.ModuleList()
-        for layer in range(self.num_layers):
-            if layer == 0:
-                self.rnn_forward.append(base_cell(input_size, hidden_size, bias, nonlinearity))
-            else:
-                self.rnn_forward.append(base_cell(hidden_size, hidden_size, bias, nonlinearity))
-
+        sizes = [input_size, *[hidden_size] * (self.num_layers - 1)]
+        self.rnn_forward = nn.ModuleList(
+            (base_cell(sz, hidden_size, bias, nonlinearity) for sz in sizes)
+        )
         if self.bidirectional:
-            self.rnn_backward = nn.ModuleList()
-            for layer in range(self.num_layers):
-                if layer == 0:
-                    self.rnn_backward.append(base_cell(input_size, hidden_size, bias, nonlinearity))
-                else:
-                    self.rnn_backward.append(
-                        base_cell(hidden_size, hidden_size, bias, nonlinearity)
-                    )
+            self.rnn_backward = nn.ModuleList(
+                (base_cell(sz, hidden_size, bias, nonlinearity) for sz in sizes)
+            )
 
-    def forward(self, x, h=None):
-        # If batch_first == True, swap axis with seq_len
+    def forward(self, x, hc=None):
+        # Treat hc[0] as the hidden state regardless of whether it is an LSTM or GRU
+        if not self.is_lstm and hc is not None:
+            hc = [hc]
+
+        # If batch_first == True, swap batch with seq_len
         # At the end of the process we swap it back to the original structure
         if self.batch_first:
-            x, h = self._swap_axis(x, h)
-
-        # If it is LSTM, get hidden and cell states
-        if h is not None and self.is_lstm:
-            h, c = h
+            x = torch.transpose(x, 0, 1)
+            if hc is not None:
```

Comment:
I think it would be clearer if hc is processed all at once (in the same place as hc = [hc]).
For instance, we could have
```
if hc is not None:
    if not self.is_lstm:
        hc = [hc]
    if self.batch_first:
       hc = [torch.transpose(item, 0, 1) for item in hc]
```
at the top of the function

Reply:
I agree, and also added comments here to clarify what `hc` represents